servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -362,12 +361,17 @@ public boolean isDirectory(long id) {
    * Returns the file id for a given path. Called via RPC, as well as internal masters.
    *
    * @param path the path to get the file id for
-   * @return the file id for a given path
-   * @throws InvalidPathException
+   * @return the file id for a given path, or -1 if there is no file at that path
    */
-  public long getFileId(TachyonURI path) throws InvalidPathException {
+  public long getFileId(TachyonURI path) {
+    // TODO(manugoyal) make sure client and master agree that -1 is an invalid file id
```

Comment:
To enforce this, you can define a constant in IdUtils, and let both client and master reference it.